src/sdk/sql_cluster_router.cc

```diff
@@ -2658,6 +2713,73 @@ hybridse::sdk::Status SQLClusterRouter::HandleDeploy(const hybridse::node::Deplo
     sp_info.set_sql(str_stream.str());
     sp_info.set_type(::openmldb::type::ProcedureType::kReqDeployment);
 
+    // extract long windows info from select_sql
+    auto iter = deploy_node->Options()->find("long_windows");
+    std::string long_window_param = "";
+    if (iter != deploy_node->Options()->end()) {
+        long_window_param = iter->second->GetExprString();
+    }
+    std::unordered_map<std::string, std::string> long_window_map;
+    if (cluster_sdk_->IsClusterMode() && !long_window_param.empty()) {
+        if (table_pair.size() != 1) {
+            return {base::ReturnCode::kError, "unsupport multi tables with long window options"};
+        }
+        std::string base_db = table_pair.begin()->first;
+        std::string base_table = table_pair.begin()->second;
+        std::vector<std::string> windows;
+        boost::split(windows, long_window_param, boost::is_any_of(","));
+        for (auto& window : windows) {
+            std::vector<std::string> window_info;
+            boost::split(window_info, window, boost::is_any_of(":"));
+
+            if (window_info.size() == 2) {
+                long_window_map[window_info[0]] = window_info[1];
+            } else if (window_info.size() == 1) {
+                long_window_map[window_info[0]] = FLAGS_mini_window_size;
+            } else {
+                return {base::ReturnCode::kError, "illegal long window format"};
+            }
+        }
+        auto long_window_infos = base::DDLParser::ExtractLongWindowInfos(select_sql, long_window_map);
```

Comment:
return empty long_window_infos if Extract failed. it need to check `long_window_infos` here